### ISSUE TITLE
Consider adding node-fetch/register to add global fetch in Node

### ISSUE THREAD

**Issue by PatrickShaw** [node-fetch/node-fetch#763]
Might be nice to have an import that allows globally declares `fetch` for Node.

E.g. `require(node-fetch/register)` or `import 'node-fetch/register';`
Would allow you to use `fetch(...)` without having to import `node-fetch` in every file you want to use it in.

Putting it in it's own `register.js` file would also make polluting the global var an opt-in choice.

An example of what `register.js` might look like:
```js
// register.js
const fetch = require('node-fetch');

if (typeof globalThis ===  'object') {
  // See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis
  globalThis.fetch = fetch;
} else if (typeof global === 'object )' {
  // For Node <12
  globa.fetch = fetch;
} else {
  // Everything else is not supported
  throw new Error('Unknown JavaScript environment: Not supported');
}
```

Some previous discussion here by the looks of it: #60

**Comment by JefferyHus** [comment]
Using the  `global` variable is considered a bad practice for many reasons. So I don't see a problem of using `module.exports`. In the end not all your files will require to use `fetch`.

What I would got for in your case, is to look up my application architecture, I'd use the onion architecture logic to initiate the `fetch` variable and pass it on to the next `function`.

**Comment by jimmywarting** [comment]
@JefferyHus I'm just curios what this means:

> I'd use the onion architecture logic to initiate the fetch variable

**Comment by PatrickShaw** [comment]
Yes, defining things on `globalThis` (it's alias being `global` in Node) is a considered bad practice for many reasons, particularly without using Symbols. The most obvious one being that another package may already be using the key you defined on the `global` object.

However, `fetch` is defined on `globalThis` (AKA `window`) on browsers. I don't think I've ever seen browser code import `fetch` from anywhere... Maybe someone's using it as a [ponyfill](https://www.google.com/search?q=ponyfill&oq=ponyfill+&aqs=chrome..69i57j0l6j69i60.3172j0j7&sourceid=chrome&ie=UTF-8) for their app.

As such, generally speaking, in order for a source file that calls `fetch` to be shareable between Node and browser code, the simplest solution is just to define `fetch` on `globalThis`.

This is why I think, in this case, it is not necessarily bad practice to define `fetch` on `globalThis`. Note that this would be a opt-in feature. You'd specifically have to import `node-fetch/register` which would not be exported from the main entry point.

**Comment by ZeeD** [comment]
FWIW I also would find useful to have a "register.js" file in the project: in my case it's a (nested) dependency that requires fetch, and I would like to avoid changing my source code to pollute myself. Instead I would prefer to relies on (ts-)node `--require` parameter.

Other projects, like  node-localstorage, also, offer the same functionality

**Comment by PatrickShaw** [comment]
Just a heads up to everyone.
The need for `node-fetch` should no longer be required as of Node 18.

See: https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch
> No longer behind --experimental-global-fetch CLI flag.

or the actual announcement for Node 18: https://nodejs.org/en/blog/announcements/v18-release-announce#fetch-experimental
> In Node.js 18, an experimental global fetch API is available by default.

**Comment by ZeeD** [comment]
good for everyone who can already upgrade :)
in my projects I'm stuck with node 16 (it will live until [2023-09-11](https://nodejs.org/en/blog/announcements/nodejs16-eol/))
